src/app/Fake.Runtime/SdkAssemblyResolver.fs

```diff
@@ -149,6 +149,89 @@ type SdkAssemblyResolver(logLevel: Trace.VerboseLevel) =
             |> Option.bind resolveFile
             |> Option.map (fun dotnetRoot -> dotnetRoot.Directory.FullName)
 
+    /// <summary>
+    /// provides the path to the `dotnet` binary running this library, respecting various dotnet <see href="https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-environment-variables#dotnet_root-dotnet_rootx86%5D">environment variables</see>.
+    /// Also probes the PATH and checks the default installation locations
+    /// </summary>
+    member this.ResolveDotNetRoots() =
+        let isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+        let isMac = RuntimeInformation.IsOSPlatform(OSPlatform.OSX)
+        let isLinux = RuntimeInformation.IsOSPlatform(OSPlatform.Linux)
+        let isUnix = isLinux || isMac
+
+        let potentialDotnetHostEnvVars =
+            [ "DOTNET_HOST_PATH", id // is a full path to dotnet binary
+              "DOTNET_ROOT", (fun s -> Path.Combine(s, this.DotNetBinaryName)) // needs dotnet binary appended
+              "DOTNET_ROOT(x86)", (fun s -> Path.Combine(s, this.DotNetBinaryName)) ] // needs dotnet binary appended
+
+        let existingEnvVarValue envVarValue =
+            match envVarValue with
+            | null
+            | "" -> None
+            | other -> Some other
+
+        let tryFindFromEnvVar () =
+            potentialDotnetHostEnvVars
+            |> List.choose (fun (envVar, transformer) ->
+                match Environment.GetEnvironmentVariable envVar |> existingEnvVarValue with
+                | Some varValue -> Some(transformer varValue |> FileInfo)
+                | None -> None)
```

Comment:
```suggestion
        let existingEnvVarValue envVarValue =
            match envVarValue with
            | null
            | "" -> ValueNone
            | other -> ValueSome other

        let tryFindFromEnvVar () =
            potentialDotnetHostEnvVars
            |> List.choose (fun struct (envVar, transformer) ->
                Environment.GetEnvironmentVariable envVar
                |> existingEnvVarValue
                |> ValueOption.map (transformer >> FileInfo)
                |> ValueOption.toOption
           )
```

Reply:
Makes sense, this was just copied from the old code, but I'll update both.

Reply:
We loose the whole benefit of ValueTuple vs Tuple if we still have to assign both via `ValueOption.toOption` so I think this change will make sense if List.choose and List.tryPick etc will support natively ValueTuple, but if not, there is no point, just extra-mapping of types.